plugins/GSdx/Renderers/Common/GSOsdManager.cpp

```diff
@@ -397,12 +408,73 @@ size_t GSOsdManager::GeneratePrimitives(GSVertexPT1* dst, size_t count) {
 		for(const auto &pair : m_monitor) {
 			if((pair.first.size() + pair.second.size()) * 6 > count - drawn) break;
 
-			// Calculate where to start rendering from by taking the right most position 1.0
-			// and subtracting (going left) 8 scaled pixels for a margin, then subtracting
-			// the size of the longest key and subtracting a scaled space and finally
-			// subtracting the longest value
-			float x = 1.0f - 8 * (2.0f/m_real_size.x) - first_max - m_char_info[' '].ax * (2.0f/m_real_size.x) - second_max;
-			float y = -1.0f + ((m_size+2)*(2.0f/m_real_size.y)) * line++;
+			float x, y;
+			/*
+				If you decide to update this, here are some tips to get you started, would've been helpful for me
+
+				m_real_size is the size of the emulator screen, measured in pixels
+				m_size is the fontsize configured by the user
+				first_max is the largest key in pixels 
+				second_max is the largest value in pixels
+
+				The min and max value of X and Y is -1 and 1
+				When finding the X or Y value of an osd draw, it always needs a relative point on the screen, those are
+					-1.0f    (left, top)
+					0        (center)
+					1.0f     (right,bottom)
+
+				You can witness this below, the number directly after the '=' operator is a relative point on the screen.
+				
+				Whenever you draw on the left or right side of the screen, add a margin of 8 pixels
+				[ (2.0f/ m_real_size.x) * 8 ] provides you with 8 pixels scaled to the screens width 
+
+				When you see
+				[ (m_size + 2) ] it simply adds some upper and lower padding 
+
+				The X centering formula centers only the longest line, all other lines start at said longest line
+				For example, If "FPS: 120" is the longest line, and its x value is 0.3, all other lines will have x value 0.3
+
+				Hopefully this'll help the next person trying to update this
+			*/
+
+			switch (m_monitor_pos) {
+				case gsosd_pos::TOPLEFT:
```

Comment:
I suggest
```cpp
case OSDPOS::TOPLEFT:
default:
  // code
```

I understand input is sanitized but if there is a bug or a developer working on this makes a mistake debugging undefined behavior is not fun as `x` and `y` are not initialized.